src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -446,16 +443,83 @@ public async Task RemoveAccountAsync(ApplicationConfiguration appConfig, IAccoun
             }
         }
 
-        public Task<IReadOnlyList<IAccount>> GetAccountsAsync(
+        public async Task<IReadOnlyList<IAccount>> GetAccountsAsync(
             string clientID,
             string redirectUri,
             AuthorityInfo authorityInfo,
             ICacheSessionManager cacheSessionManager,
             IInstanceDiscoveryManager instanceDiscoveryManager)
         {
-            // runtime does not yet support account discovery
-            using LogEventWrapper logEventWrapper = new LogEventWrapper(this);
-            return Task.FromResult<IReadOnlyList<IAccount>>(Array.Empty<IAccount>());
+            if (!_wamOptions.ListWindowsWorkAndSchoolAccounts)
+            {
+                _logger.Info("[WamBroker] ListWindowsWorkAndSchoolAccounts option was not enabled.");
+                return Array.Empty<IAccount>();
+            }
+
+            Debug.Assert(s_lazyCore.Value != null, "Should not call this API if msal runtime init failed");
+
+            var requestContext = cacheSessionManager.RequestContext;
+
+            using (var discoverAccountsResult = await s_lazyCore.Value.DiscoverAccountsAsync(
+                clientID,
+                cacheSessionManager.RequestContext.CorrelationId.ToString("D"), 
+                cacheSessionManager.RequestContext.UserCancellationToken).ConfigureAwait(false))
+            {
+                if (discoverAccountsResult.IsSuccess)
+                {
+                    List<NativeInterop.Account> wamAccounts = discoverAccountsResult.Accounts;
+
+                    _logger.Verbose($"[WamBroker] Broker returned {wamAccounts.Count()} account(s).");
```

Comment:
What about making this Info level? If we're debugging an issue with Info level logs only, would probably help to know if we returned broker accounts, unless we log some this info somewhere upstream.